vllm/v1/spec_decode/medusa.py

```diff
@@ -42,11 +42,18 @@ def propose(
         blocks = self.model(target_hidden_states)
         logits = self.model.compute_logits(blocks)
 
-        # Get draft tokens and transpose the result
-        # TODO(woosuk): OPTIMIZATION: Return GPU tensor without GPU-CPU
-        # synchronization.
-        draft_tokens = [logit.argmax(dim=-1).tolist() for logit in logits]
-        return [list(row) for row in zip(*draft_tokens)]
+        # Non-primary DP/TP ranks emit an empty placeholder.
```

Comment:
What commands did you run to compare the effects of this change, in terms of correctness and/or performance?

Reply:
Please run an end-to-end test serving an actual model, and benchmark the performance before and after. Ideally, you could [confirm with nsight systems](https://docs.vllm.ai/en/latest/contributing/profiling.html) that this does reduce the overhead.